File: LeetCode/204. Count Primes/204.c
#include <stdio.h>
#include <stdlib.h>

void RemoveMultiples(int* primes,int multple,int size){
    for(int i = multple*2; i < size; i+=multple ){
        primes[i] = 0;
    }
}

int countPrimes(int n) {
    int Resolve = 0;

    int *primes =(int*)malloc(n* sizeof(int));
    for (int i = 0; i < n; i++) {
        primes[i] = 1;
    }
    for(int i = 2; i<n; i++){
        if(primes[i]){
            Resolve+=1;
            RemoveMultiples(primes, i, n);
        }
    }

    return Resolve;
}

File: LeetCode/204. Count Primes/204.py
class Solution:

    def NewInit(self, Temporary: int) -> None:
        self.primes = [True for _ in range(Temporary)]
        self.qtd = Temporary
        self.Resolve = 0


    def RemoveMultiple(self, multiple: int) -> None:
        for i in range(multiple*2 , self.qtd, +multiple):
            self.primes[i] = False

    def countPrimes(self, n: int) -> int:
        self.NewInit(n)
        for i in range(2, n): 
            if self.primes[i]:
                self.RemoveMultiple(i)
                self.Resolve+=1
        
        return self.Resolve


